### ISSUE TITLE
Out of Memory Exception - Dispatcher.java line 110

### ISSUE THREAD

**Issue by Fgabz** [intercom/intercom-android#197]
I just got the issue from my crashlytics logs: 

## Version info
  - Intercom for Android version: 1.1.18
  - Android OS version: 6.0.1
  - Android Device: sm_g925f

## Logs

`Fatal Exception: java.lang.OutOfMemoryError: Could not allocate JNI Env
       at java.lang.Thread.nativeCreate(Thread.java)
       at java.lang.Thread.start(Thread.java:1063)
       at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:921)
       at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1339)
       at io.intercom.com.squareup.okhttp.Dispatcher.enqueue(Dispatcher.java:110)
       at io.intercom.com.squareup.okhttp.Call.enqueue(Call.java:114)
       at io.intercom.com.squareup.okhttp.OkHttpClient$1.callEnqueue(OkHttpClient.java:98)
       at io.intercom.com.squareup.okhttp.ws.WebSocketCall.enqueue(WebSocketCall.java:109)
       at io.intercom.android.sdk.nexus.NexusSocket$2.run(NexusSocket.java:77)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:158)
       at android.os.HandlerThread.run(HandlerThread.java:61)`

**Comment by ronocod** [comment]
Hi @Fgabz,

Thanks for reporting this, we've fixed this bug and the fix will be in the next version. I'll reply here when it's released.

**Comment by harkin** [comment]
Hey, thanks for letting us know about this! I suspect this is a dupe of a long standing issue we've had: https://github.com/intercom/intercom-android/issues/136

I've made some changes that will be going out in the next update (soon) that will address this issue :)

**Comment by Digisha** [comment]
We got the same crash in our app:

- Intercom for Android version: 1.1.21
- Android OS version: 6.0.1
- Android Device: SM-G920R4, SM-G930R4

```
java.lang.OutOfMemoryError: Could not allocate JNI Env
1	at java.lang.Thread.nativeCreate(Native Method)
2	at java.lang.Thread.start(Thread.java:1063)
3	at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:921)
4	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1339)
!	5	at io.intercom.com.squareup.okhttp.Dispatcher.enqueue(Dispatcher.java:110)
6	at io.intercom.com.squareup.okhttp.Call.enqueue(Call.java:114)
7	at io.intercom.com.squareup.okhttp.OkHttpClient$1.callEnqueue(OkHttpClient.java:98)
8	at io.intercom.com.squareup.okhttp.ws.WebSocketCall.enqueue(WebSocketCall.java:109)
9	at io.intercom.android.sdk.nexus.NexusSocket$2.run(NexusSocket.java:77)
10	at android.os.Handler.handleCallback(Handler.java:739)
11	at android.os.Handler.dispatchMessage(Handler.java:95)
12	at android.os.Looper.loop(Looper.java:158)
13	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by ronocod** [comment]
We've released version [3.0.0](https://github.com/intercom/intercom-android/releases/tag/3.0.0) which fixes this by using less threads and limiting the number of threads we create. If you still see this issue after updating to 3.0.0 please let us know.
